exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -103,6 +103,25 @@ def shutdown(timeout: nil)
 
         private
 
+        # Builds span flags based on whether the parent span context is remote.
+        # This follows the OTLP specification for span flags.
+        def build_span_flags(parent_span_is_remote, base_flags)
+          # Extract integer value from TraceFlags object if needed
+          base_flags_int =
+            if base_flags.is_a?(OpenTelemetry::Trace::TraceFlags)
+              base_flags.instance_variable_get(:@flags)
+            else
+              base_flags
+            end
```

Comment:
This is problematic. We should not reach into the internals of `TraceFlags` to retrieve the flags. We should use the `sampled?` method and build the flags accordingly.
```suggestion
          base_flags_int =
            if base_flags.sampled?
              1
            else
              0
            end
```

Is there a case where this method is called with an Integer for `base_flags` instead of a `TraceFlags`? There shouldn't be.

Reply:
done

Reply:
Check this failing workflow error please @fbogsany 
```
Command failed with status (1)
  Finished in 1.297985s, 42.3734 runs/s, 106.3186 assertions/s.
  /Users/runner/hostedtoolcache/Ruby/3.4.1/arm64/bin/bundle:25:in 'Kernel#load'
  
  /Users/runner/hostedtoolcache/Ruby/3.4.1/arm64/bin/bundle:25:in '<main>'
    1) Error:
  Tasks: TOP => test
  OpenTelemetry::Exporter::OTLP::Exporter::span flags::build_span_flags#test_0001_sets flags to HAS_IS_REMOTE for local parent span context:
  (See full trace by running task with --trace)
  NoMethodError: undefined method 'sampled?' for an instance of Integer
      lib/opentelemetry/exporter/otlp/exporter.rb:112:in 'OpenTelemetry::Exporter::OTLP::Exporter#build_span_flags'
      test/opentelemetry/exporter/otlp/exporter_test.rb:817:in 'block (4 levels) in <top (required)>'
  
    2) Error:
  OpenTelemetry::Exporter::OTLP::Exporter::span flags::build_span_flags#test_0002_sets flags to HAS_IS_REMOTE | IS_REMOTE for remote parent span context:
  NoMethodError: undefined method 'sampled?' for an instance of Integer
      lib/opentelemetry/exporter/otlp/exporter.rb:112:in 'OpenTelemetry::Exporter::OTLP::Exporter#build_span_flags'
      test/opentelemetry/exporter/otlp/exporter_test.rb:824:in 'block (4 levels) in <top (required)>'
  
    3) Error:
  OpenTelemetry::Exporter::OTLP::Exporter::span flags::build_span_flags#test_0003_preserves base trace flags:
  NoMethodError: undefined method 'sampled?' for an instance of Integer
      lib/opentelemetry/exporter/otlp/exporter.rb:112:in 'OpenTelemetry::Exporter::OTLP::Exporter#build_span_flags'
      test/opentelemetry/exporter/otlp/exporter_test.rb:832:in 'block (4 levels) in <top (required)>'
  
  55 runs, 138 assertions, 0 failures, 3 errors, 1 skips
  
  You have skipped tests. Run with --verbose for details.
  Coverage report generated for Unit Tests to /Users/runner/work/opentelemetry-ruby/opentelemetry-ruby/exporter/otlp/coverage.
  Line Coverage: 90.27% (269 / 298)
  ```